logstash-core/src/main/java/org/logstash/Logstash.java

```diff
@@ -61,6 +63,15 @@ public static void main(final String... args) {
         System.exit(0);
     }
 
+    private static void configureNashornDeprecationSwitchForJavaAbove11() {
+        final String javaVersion = System.getProperty("java.version");
+        // match version 1.x.y, 9.x.y and 10.x.y
+        if (!javaVersion.matches("^1\\.\\d\\..*") && !javaVersion.matches("^(9|10)\\.\\d\\..*")) {
```

Comment:
wouldn't this work as well?

```suggestion
        if javaVersion.matches("^11.*") {
```

Reply:
or maybe `1[1-4]`, but we currently only support java 8 and 11 anyway

Reply:
We know that the flag is not available in java 1.x (1.8) and 9.x, 10.x, it's in 11.x an above but don't know upfront for how many version they keep the deprecation flag before dropping the Nashorn. It seems an approach more future proof I think.